### PR TITLE
Switching to using apachesolr module instead of Search API

### PR DESCRIPTION
**Pull request by blisteringherb**
- Apachesolr module properly indexes all fields for all types

### CODE REVIEW

File: lib/modules/dosomething/dosomething_search/dosomething_search.apachesolr_environments.inc [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * Implements hook_apachesolr_environments().
+ */
+function dosomething_search_apachesolr_environments() {
+  $export = array();
+
+  $environment = new stdClass();
+  $environment->api_version = 1;
+  $environment->env_id = 'solr';
+  $environment->name = 'localhost server';
+  $environment->url = 'http://192.168.1.21:8080/solr/collection1';
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_search/dosomething_search.apachesolr_environments.inc]: I'd imagine we'd want to store the name and URL outside of code, no?  When I go that URL I get "The requested resource is not available."

File: lib/modules/dosomething/dosomething_search/dosomething_search.apachesolr_environments.inc [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  $environment->conf = array(
+    'apachesolr_direct_commit' => 0,
+    'apachesolr_index_updated' => 1392910246,
+    'apachesolr_last_optimize' => 1392910246,
+    'apachesolr_read_only' => '0',
+    'apachesolr_soft_commit' => 0,
+  );
+  $environment->index_bundles = array(
+    'node' => array(
+      0 => 'fact',
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_search/dosomething_search.apachesolr_environments.inc]: I don't believe we want fact nodes to show up in search.  The fact node just exists for internal usage (so a user would get an access denied upon viewing a fact/[nid] page.

**Comment by aaronschachter** [comment]
This looks good, although I wonder if it's going to break things without a new build, as the search module needs a new module to be downloaded via the build script.  @mshmsh5000 @desmondmorris -- thoughts?  I'd imagine that the search feature would just be disabled...

**Comment by mshmsh5000** [comment]
I think we need to update the Jenkins build to run drush make into a temp directory, then switch that with the live html dir. Like

`drush make /vagrant/build-dosomething.make html-new`

and so on. Need to redo the symlinks to settings and files, etc.

**Comment by mshmsh5000** [comment]
To be clear -- Jenkins-side changes shouldn't block this PR.

**Comment by aaronschachter** [comment]
ok, sounds good :+1:

**Pull request merged**